dns/bind/src/opnsense/service/templates/OPNsense/Bind/named.conf

```diff
@@ -186,10 +186,17 @@ logging {
                 print-time yes;
         };
 
-        category default { default_log; };
-        category general { default_log; };
-        category queries { query_log; };
-        category rpz { rpz_log; };
+{% if helpers.exists('OPNsense.bind.general.syslog') %}
+        channel syslog {
+                syslog daemon;
+                severity info;
+        };
+{% endif %}
```

Comment:
And move trailing iff below newly added lines?

Reply:
I assume you mean moving `{% endif %}` below line 199? That would mean a re-write of the whole section. We only want the additional `syslog` option for each `category` if the `general.syslog` helper is enabled.

Quick, completely untested and maybe even non-working example:

```
{% if helpers.exists('OPNsense.bind.general.syslog') %}
        channel syslog {
                syslog daemon;
                severity info;
        };
        category default { default_log;{{ ' syslog;' if 'default' in OPNsense.bind.general.syslog|string }} };
        category general { default_log;{{ ' syslog;' if 'general' in OPNsense.bind.general.syslog|string }} };
        category queries { query_log;{{ ' syslog;' if 'queries' in OPNsense.bind.general.syslog|string }} };
        category rpz { rpz_log;{{ ' syslog;' if 'rpz' in OPNsense.bind.general.syslog|string }} };
{% else %}
        category default { default_log; };
        category general { default_log; };
        category queries { query_log; };
        category rpz { rpz_log; };
{% endif %}